pkg/apis/provisioning.cattle.io/v1/rke.go

```diff
@@ -77,6 +78,11 @@ type RKEConfig struct {
 	RotateCertificates   *rkev1.RotateCertificates   `json:"rotateCertificates,omitempty"`
 	RotateEncryptionKeys *rkev1.RotateEncryptionKeys `json:"rotateEncryptionKeys,omitempty"`
 
-	MachinePools      []RKEMachinePool        `json:"machinePools,omitempty"`
-	InfrastructureRef *corev1.ObjectReference `json:"infrastructureRef,omitempty"`
+	MachinePools        []RKEMachinePool        `json:"machinePools,omitempty"`
+	MachinePoolDefaults RKEMachinePoolDefaults  `json:"machinePoolDefaults,omitempty"`
```

Comment:
Question: Why do we define a new struct instead of adding `HostnameLengthLimit int`  directly here? Is it because we must distinguish between zero value and "unset"?

Reply:
Got it. sounds good to me.

Reply:
it's an organizational/future proofing thing. in the future we may want other default machinepool values